src/main/java/org/desarrolladorslp/technovation/services/impl/MessageServiceImpl.java

```diff
@@ -53,26 +55,68 @@ public MessageBodyDTO getSpecificMessageByUser(UUID messageId, UUID userReceiver
 
     @Override
     @Transactional
-    public void markMessageAsRead(UUID messageId) {
-        messageRepository.markMessageAsRead(messageId);
+    public void markMessageAsRead(UUID messageId, UUID userReceiverId) {
+        Map<String, List<MessageHeaderDTO>> messages = getMessagesByUser(userReceiverId);
+        if (messages.values().stream().anyMatch(p -> p.stream().anyMatch(m -> m.getMessageId().equals(messageId)))) {
+            messageRepository.markMessageAsRead(messageId);
+        } else {
+            throw new MessageDoesNotBelongToUser(messageId + "is not owned by user");
+        }
     }
 
     @Override
     @Transactional
-    public void markMessageAsUnread(UUID messageId) {
-        messageRepository.markMessageAsUnread(messageId);
+    public void markMessageAsUnread(UUID messageId, UUID userReceiverId) {
+        Map<String, List<MessageHeaderDTO>> messages = getMessagesByUser(userReceiverId);
```

Comment:
you can use `getBodyOfMessageByUser` from `MessageRepository` instead of getMessagesByUser and it will be much more efficient. (it is already on develop)